developer-workflow/c-api.rst

```diff
@@ -134,6 +145,101 @@ When moving existing tests, feel free to replace ``TestError`` with
 ``PyExc_AssertionError`` unless actually testing custom exceptions.
 
 
+.. _unstable-capi:
+
+Unstable C API
+==============
+
+The unstable C API tier is meant for extensions that need tight integration
+with the interpreter, like debuggers and JIT compilers.
+Users of this tier may need to change their code with every minor release.
+
+In many ways, this tier is like the general C API:
+
+- it's available when ``Python.h`` is included normally,
+- it should be defined  in ``Include/cpython/``,
+- it requires tests, so we don't break it unintentionally
+- it requires docs, so both we and the users
+  can agree on the expected behavior,
+- it is tested and documented in the same way.
+
+The differences are:
+
+- Names of functions structs, macros, etc. start with the ``PyUnstable_``
+  prefix. This defines what's in the unstable tier.
+- The unstable API can change in minor versions, without any deprecation
+  period.
+- A stability note is appears in the docs.
+  This happens automatically, based on the name
+  (via :cpy-file:`Doc/tools/extensions/c_annotations.py`).
+
+Despite being “unstable”, there are rules to make sure third-party code can
+use this API reliably:
+
+* Changes and removals can be done in minor releases (3.x.0, including
+  Alphas and Betas for 3.x.0).
```

Comment:
```suggestion
* Changes and removals can be done in minor releases
  (:samp:`3.{X}.0`, including Alphas and Betas for :samp:`3.{X}.0`).
```

* Use `:samp:` to explicitly mark up and format placeholder
* Follow typical convention (at least that I've seen) to use capitals for placeholder letters in versions (with lowercase being a literal `x`)

Reply:
I'm not aware of that convention. I've always used lowercase. (And I like it because it stands out among the numbers...)
Should https://docs.python.org/3.10/c-api/stable.html be updated?

Reply:
`:samp:` looks useful! TIL. thanks!

Reply:
Hmm, maybe it's just what I've seen then—particularly since I've had to use a version placeholder in a decent number of instances that also contain a literal `x`, such as branch names (`X.Y.x`). In any case, here's a version of the suggestion with a lowercase `x`:

```suggestion
* Changes and removals can be done in minor releases
  (:samp:`3.{x}.0`, including Alphas and Betas for :samp:`3.{x}.0`).
```